core/math/a_star.h

```diff
@@ -142,4 +144,46 @@ class AStar : public Reference {
 	~AStar();
 };
 
+class AStar2D : public Reference {
+	GDCLASS(AStar2D, Reference);
+	AStar *astar;
```

Comment:
I think you don't need a pointer to `astar`, as you have only once instance of it, have it fully defined above, and don't return it from any of the functions.

Reply:
AStar2D is a wrapper class above AStar so keeping a pointer inside is neccessary to call the AStar functions.

Reply:
I mean, making it a `AStar astar;`, without it being a pointer. Just tested, and it works, though it is probably not the best thing to do.

Reply:
Ah, ok